core/src/event_id.rs

```diff
@@ -146,6 +146,12 @@ impl From<&Vec<u8>> for EventId {
     }
 }
 
+impl From<&str> for EventId {
+    fn from(s: &str) -> Self {
+        EventId(s.as_bytes().to_vec())
```

Comment:
A UTF-8 string is not a valid EventId, so unless we are decoding the string as some binary representation I don't think we should have a From<str> at all.